analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -145,19 +146,25 @@ private static bool HasMutableUsagesInMethod(VariableDeclaratorSyntax parameter,
                 || arg is LambdaExpressionSyntax;
 
             bool MatchesIdentifier(IdentifierNameSyntax id) =>
-                Equals(parameterSymbol, semanticModel.GetSymbolInfo(id).Symbol);
+                Equals(variableSymbol, semanticModel.GetSymbolInfo(id).Symbol);
         }
 
         private static bool IsMutatingUse(IdentifierNameSyntax identifier) =>
             identifier.Parent switch
             {
                 AssignmentExpressionSyntax assignmentExpression => Equals(identifier, assignmentExpression?.Left),
-                ArgumentSyntax argumentSyntax => !argumentSyntax.RefOrOutKeyword.IsKind(SyntaxKind.None),
+                ArgumentSyntax argumentSyntax => IsAssignmentToTuple(argumentSyntax) || !argumentSyntax.RefOrOutKeyword.IsKind(SyntaxKind.None),
                 PostfixUnaryExpressionSyntax _ => true,
                 PrefixUnaryExpressionSyntax _ => true,
                 _ => false
             };
 
+        // (arg, b) = something
+        private static bool IsAssignmentToTuple(ArgumentSyntax argument) =>
+            TupleExpressionSyntaxWrapper.IsInstance(argument.Parent)
+            && argument.Parent.Parent is AssignmentExpressionSyntax assignment
+            && assignment.Left == argument.Parent;
```

Comment:
here, I think that it would be nicer to avoid the arrow method and store `argument.Parent` in a `var tupleExpression` - it would be easier to read